### PR TITLE
SDRClassifier Development

### LINKED ISSUE #438: HTM.java needs a SDRClassifier
HTM.java needs a [SDRClassifier](https://github.com/numenta/nupic/blob/master/src/nupic/algorithms/sdr_classifier.py#L35) (and a KNNClassifier) to have a full set of classifiers. #132 addresses the need for a KNNClassifier. This issue does the same, but for the SDRClassifier.

### PR DESCRIPTION
**Pull request by Hopding**
Fixes #438. 
Please review this PR, and suggest improvements to allow full integration of the SDRClassifier into HTM.java! :smiley:

### CODE REVIEW

**Comment by cogmission** [comment]
@Hopding Hopefully coming today... Please bear with me... ;-)

**Comment by cogmission** [comment]
@Hopding Please sync your fork with the latest changes from HTM.Java master, (resolve any conflicts, though I doubt they'll be any) then merge them into your ```sdr-classifier-dev``` branch; commit and push them back up so this PR will be clear and I can look at it?

If you don't know how to sync your fork, please [follow the directions here.](https://help.github.com/articles/syncing-a-fork/)

Thanks! :-)

**Comment by Hopding** [comment]
There was a small conflict within the `.classpath` file, but I've fixed that now :grinning:

File: src/main/java/org/numenta/nupic/algorithms/SDRClassifier.java [hunk 1]
```diff
@@ -0,0 +1,504 @@
...
+import java.util.List;
+import java.util.Map;
+
+import no.uib.cipr.matrix.sparse.FlexCompRowMatrix;
+import org.numenta.nupic.Persistable;
+import org.numenta.nupic.util.ArrayUtils;
+import org.numenta.nupic.util.Deque;
+import org.numenta.nupic.util.Tuple;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
```
Comment (cogmission) [inline src/main/java/org/numenta/nupic/algorithms/SDRClassifier.java]: Please remove this unused import:
```import com.fasterxml.jackson.databind.ObjectMapper;```

File: src/main/java/org/numenta/nupic/algorithms/SDRClassifier.java [hunk 2]
```diff
@@ -0,0 +1,504 @@
...
+	 * @param patternNZ int[] of the active indices from the output below
+	 * @param classification {@link Map} of the classification information:
+	 * <p>&emsp;"bucketIdx" - index of the encoder bucket
+	 * <p>&emsp;"actValue" -  actual value doing into the encoder
+     * @return
+	 * {@link Classification} containing inference results. The Classification
+	 * contains the computed probability distribution (relative likelihood for each
+	 * bucketIdx starting from bucketIdx 0) for each step in {@code steps}. The
+	 * Classification also contains the average actual value for each bucket.
+     */
```
Comment (cogmission) [inline src/main/java/org/numenta/nupic/algorithms/SDRClassifier.java]: Please add the suppression of unchecked casts here:
```@SuppressWarnings("unchecked")```

**Comment by cogmission** [comment]
👍 
@Hopding 
Andrew, 

**You've done an amazing job here**, I'm very happy with the results! 
Once you've made the 2 above changes we can merge this!
...and then I'll make an announcement!

**Pull request merged**